VideoLocker/src/main/java/org/edx/mobile/profiles/UserProfilePresenter.java

```diff
@@ -0,0 +1,147 @@
+package org.edx.mobile.profiles;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.text.TextUtils;
+
+import org.edx.mobile.logger.Logger;
+import org.edx.mobile.model.api.ProfileModel;
+import org.edx.mobile.module.analytics.ISegment;
+import org.edx.mobile.module.prefs.UserPrefs;
+import org.edx.mobile.user.Account;
+import org.edx.mobile.user.UserAPI;
+import org.edx.mobile.util.InvalidLocaleException;
+import org.edx.mobile.util.LocaleUtils;
+import org.edx.mobile.util.observer.Observer;
+import org.edx.mobile.view.ViewHoldingPresenter;
+
+import de.greenrobot.event.EventBus;
+
+public class UserProfilePresenter extends ViewHoldingPresenter<UserProfilePresenter.ViewInterface> {
+
+    @NonNull
+    private final String username;
+
+    private final boolean isViewingOwnProfile;
+
+    @NonNull
+    private final UserProfileInteractor userProfileInteractor;
+
+    protected final Logger logger = new Logger(getClass().getName());
+
+    public UserProfilePresenter(@NonNull final String username, @NonNull UserPrefs userPrefs, @NonNull ISegment segment, @NonNull EventBus eventBus, @NonNull UserAPI userAPI) {
+        this.username = username;
+        this.userProfileInteractor = new UserProfileInteractor(username, userAPI, eventBus);
+
+        final ProfileModel model = userPrefs.getProfile();
+        isViewingOwnProfile = null != model && model.username.equalsIgnoreCase(username);
+
+        segment.trackProfileViewed(username);
+    }
+
+    @Override
+    public void attachView(@NonNull final ViewInterface view) {
+        super.attachView(view);
+        view.setUsername(username);
+        view.setEditProfileMenuButtonVisible(isViewingOwnProfile);
+        view.showLoadingIndicator();
+        observeOnView(userProfileInteractor.observeAccount()).subscribe(new Observer<Account>() {
+            @Override
+            public void onData(@NonNull Account account) {
+                if (account.requiresParentalConsent() || account.getAccountPrivacy() == Account.Privacy.PRIVATE) {
```

Comment:
@aleffert I might prefer more methods rather than a concrete "ViewModel". But I gave it a try. Is this what you are imagining? https://github.com/edx/edx-app-android/commit/09ad1645cccb20dba0ce73334d9cc7787fbffc01